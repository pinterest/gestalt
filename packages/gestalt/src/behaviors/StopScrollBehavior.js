/* @flow */
import * as React from 'react';

type Props = {|
  children: React.Node,
|};

export default class NoScrollBehavior extends React.Component<Props> {
  constructor(props: Props) {
    super(props);
    this.prevOverflow = null;
  }

  componentDidMount() {
    if (typeof window !== 'undefined') {
      this.prevOverflow = window.document.body.style.overflow;
      window.document.body.style.overflow = 'hidden';
    }
  }

  componentWillUnmount() {
    if (typeof window !== 'undefined') {
      window.document.body.style.overflow = this.prevOverflow;
    }
  }

  prevOverflow: string | null;

  render() {
    return this.props.children;
  }
}
