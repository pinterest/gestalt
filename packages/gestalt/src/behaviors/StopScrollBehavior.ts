import { Component, ReactNode } from 'react';

type Props = {
  children: ReactNode;
};

export default class NoScrollBehavior extends Component<Props> {
  prevOverflow: string | null;

  constructor(props: Props) {
    super(props);
    this.prevOverflow = null;
  }

  componentDidMount() {
    if (typeof window !== 'undefined') {
      this.prevOverflow = window.document.body.style.overflow;
      window.document.body.style.overflow = 'hidden';
    }
  }

  componentWillUnmount() {
    if (typeof window !== 'undefined') {
      // @ts-expect-error - TS2322 - Type 'string | null' is not assignable to type 'string'.
      window.document.body.style.overflow = this.prevOverflow;
    }
  }

  render(): ReactNode {
    return this.props.children;
  }
}
