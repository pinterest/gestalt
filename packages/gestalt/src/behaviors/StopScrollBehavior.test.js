// @flow
import React from 'react';
import { create } from 'react-test-renderer';
import StopScrollBehavior from './StopScrollBehavior.js';

test('StopScrollBehavior renders', () => {
  const tree = create(
    <StopScrollBehavior>Hello world</StopScrollBehavior>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
