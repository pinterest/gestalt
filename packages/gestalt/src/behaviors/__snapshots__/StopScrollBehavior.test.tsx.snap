// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopScrollBehavior renders 1`] = `"Hello world"`;
